crates/types/src/block_impl.rs

```diff
@@ -44,10 +44,9 @@ impl VIDTransaction {
 
         for txn in transactions {
             // Encode the length of the inner transaction and the transaction bytes.
-            if let Ok(len) = txn.0.len().try_into() {
-                encoded.extend::<Vec<u8>>(vec![len]);
-                encoded.extend(txn.0);
-            }
+            let txn_size = txn.0.len().to_le_bytes();
```

Comment:
`len()` here returns a `usize`, which doesn't have the same number of bytes on every platform. Thus if we encode this way on a 64-bit machine and send this payload to a 32-bit machine, it won't deserialize properly. Better to do `(txn.0.len() as u32).to_le_bytes()` so that the length is always 4 bytes no matter what

Reply:
I thought using `size_of::<usize>()` would make the deserialization correct, but yeah, I think using `u32` is better so that we don't have to handle the platform difference.

Reply:
Are we assuming that `usize` > `u32`? If not then we shouldn't use `as u32` because that will silently wrap. (Why is it that the easy conversion using `as` is also the most dangerous?? Thanks Rust.)

Reply:
Edited previous comment. As a rule I don't like anything `usize as u32`. In this case it's probably ok because the length will never be large enough to overflow a u32. But I'd rather not rely on that.

Reply:
Thinking more about this, if we are going to use `usize as`, I think we should use `u64` instead, to make sure the length is enough.

Reply:
uh, are we willing to send an additional 4 bytes of zeros over the wire just to save us the trouble of properly converting usize to u32? If panic is unacceptable then perhaps the method could return a `Result`.

Reply:
What's the purpose of this `encode` method? I'm not aware of any reason why hotshot should flatten a vec of tx bytes like this. Indeed, I it's not clear to me that anything in this file needs to exist. See my [other comment](https://github.com/EspressoSystems/HotShot/pull/2045#discussion_r1389992991)

Reply:
Ah, this is a good point. Since Everything in this file is non-production code (but does need to exist), I think panic should be acceptable and there's no need to use u64.

Reply:
> (but does need to exist for HotShot tests)

In that case, is it ok to move everything here into test code?

Reply:
We'll still have this problem in the sequencer, though. I think it is probably ok for `encode` to return a `Result`. It is only called in block building, so if it returns `Err` it basically means you have not built a valid block. In this case the leader could try a different builder or just not proposer. However, it might be slightly better to have `from_transactions` return a `Result`, so you do all the necessary error checking there and then the rest of the functions are pure.

I hate to say it, but we could also bring back the incremental `add_transaction` API, so that you can reject individual transactions (e.g. if their size is more than 2^32 bytes). But that's more of an optimization, we definitely don't need it now.

Reply:
>In that case, is it ok to move everything here into test code?

Yes! I've created an [issue](https://github.com/EspressoSystems/HotShot/issues/2059). Will work on that next.

> However, it might be slightly better to have from_transactions return a Result
Yeah, I think this is better. Will update `encode` and `from_transactions` (not bringing back `add_transaction` for now).